training/train.py

```diff
@@ -302,30 +302,48 @@ def create_git_pr(s3_client, model_version, recall_at_30):  # pragma: no cover
     prev_hyperparams = s3_client.read_json_file(k)
 
     # Convert the json description to string
-    description = json.dumps(prev_hyperparams).replace('"', '\\"')
+    description = 'Previous hyper parameters :: ' + json.dumps(prev_hyperparams).replace('"', '\\"')
 
     prev_recall = prev_hyperparams.get('recall_at_30', 0.55)
+    _logger.info('create_git_pr:: Prev => Model {}, Recall {}  Curr => Model {}, Recall {}'.format(
+        previous_version, prev_recall, model_version, recall_at_30))
     if recall_at_30 >= prev_recall:
         try:
             # Invoke bash script to create a saas-analytics PR
-            t = subprocess.Popen(['sh', 'rudra/utils/github_helper.sh', 'f8a-hpf-insights.yaml',
-                                  'MODEL_VERSION', str(model_version), description],
-                                 shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            # Wait for the subprocess to get over
-            t.wait(60)
-            if t.returncode == 0:
-                _logger.info("Successfully created a PR")
+            t1 = subprocess.Popen(['wget', '-v',
+                                   'https://raw.githubusercontent.com/fabric8-analytics/'
+                                   'fabric8-analytics-rudra/master/rudra/utils/github_helper.sh',
+                                   '-O', './github_helper.sh'], shell=False)
+            t1.wait(60)
+            _logger.info('Getting github helper, return code: {}'.format(t1.returncode))
+
+            t2 = subprocess.Popen(['chmod', '+x', './github_helper.sh'], shell=False)
+            t2.wait(60)
+            _logger.info('Setting exec permission to helper, return code: {}'.format(
+                t2.returncode))
+
+            t3 = subprocess.Popen(['./github_helper.sh', 'f8a-pypi-insights.yaml',
+                                   'MODEL_VERSION', str(model_version), description], shell=False)
+            t3.wait(60)
+            if t3.returncode == 0:
+                _logger.info("Successfully created a SAAS PR.")
```

Comment:
Are we creating saas PR? first it should be deployed on staging right?

Reply:
Even to deploy on stag we need a SAAS PR to update stag model value. In fact there is issue in current code which does not differentiate between stag and prod, I am working on that under jira ticket, https://issues.redhat.com/browse/APPAI-1733 I want to avoid too many changes in one PR which make it difficult to understand and move.